core/src/main/resources/defaults/blueprints/7.2.17/cdp-sdx-scalable.bp

```diff
@@ -0,0 +1,516 @@
+{
+  "tags":{
+    "shared_services_ready":true
+  },
+  "description":"7.2.17 - Scalable SDX template with Atlas, HMS, Ranger and other services they are dependent on.  Services like HDFS, HBASE, RANGER, HMS have HA",
+  "blueprint":{
+    "cdhVersion":"7.2.17",
+    "displayName":"scalable-datalake",
+    "hostTemplates":[
+      {
+        "cardinality":2,
+        "refName":"master",
+        "roleConfigGroupsRefNames":[
+          "zookeeper-SERVER-BASE",
+          "hbase-MASTER-BASE",
+          "ranger-RANGER_ADMIN-BASE",
+          "hdfs-GATEWAY-BASE",
+          "hive-GATEWAY-BASE",
+          "hbase-GATEWAY-BASE",
+          "solr-GATEWAY-BASE",
+          "kafka-GATEWAY-BASE",
+          "hdfs-NAMENODE-BASE",
+          "hdfs-FAILOVERCONTROLLER-BASE",
+          "hdfs-JOURNALNODE-BASE"
+        ]
+      },
+      {
+        "cardinality":2,
+        "refName":"gateway",
+        "roleConfigGroupsRefNames":[
+          "atlas-ATLAS_SERVER-BASE",
+          "ranger-RANGER_ADMIN-BASE",
+          "ranger-RANGER_USERSYNC-BASE",
+          "hive-HIVEMETASTORE-BASE",
+          "knox-KNOX_GATEWAY-BASE",
+          "solr-GATEWAY-BASE",
+          "hdfs-GATEWAY-BASE",
+          "kafka-GATEWAY-BASE",
+          "hive-GATEWAY-BASE",
+          "hbase-GATEWAY-BASE"
+        ]
+      },
+      {
+        "cardinality":3,
+        "refName":"core",
+        "roleConfigGroupsRefNames":[
+          "hdfs-DATANODE-BASE",
+          "kafka-KAFKA_BROKER-BASE",
+          "solr-SOLR_SERVER-BASE",
+          "hbase-REGIONSERVER-BASE",
+          "solr-GATEWAY-BASE",
+          "hdfs-GATEWAY-BASE",
+          "kafka-GATEWAY-BASE",
+          "hive-GATEWAY-BASE",
+          "hbase-GATEWAY-BASE"
+        ]
+      },
+      {
+        "cardinality":1,
+        "refName":"auxiliary",
+        "roleConfigGroupsRefNames":[
+          "ranger-RANGER_TAGSYNC-BASE",
+          "zookeeper-SERVER-BASE",
+          "hdfs-JOURNALNODE-BASE"
+        ]
+      },
+      {
+        "cardinality":2,
+        "refName":"idbroker",
+        "roleConfigGroupsRefNames":[
+          "knox-IDBROKER-BASE"
+        ]
+      },
+      {
+        "cardinality":0,
+        "refName":"solrHG",
+        "roleConfigGroupsRefNames":[
+          "solr-SOLR_SERVER-BASE"
+        ]
+      },
+      {
+        "cardinality":0,
+        "refName":"storageHG",
+        "roleConfigGroupsRefNames":[
+          "hbase-REGIONSERVER-BASE",
+          "hdfs-DATANODE-BASE"
+        ]
+      },
+      {
+        "cardinality":0,
+        "refName":"kafkaHG",
+        "roleConfigGroupsRefNames":[
+          "kafka-KAFKA_BROKER-BASE"
+        ]
+      },
+      {
+        "cardinality":0,
+        "refName":"razHG",
+        "roleConfigGroupsRefNames":[
+          "ranger-RANGER_ADMIN-BASE"
```

Comment:
This needs to be a RAZ reference, correct?

Also - where is RAZ configured - injected via code? Which RG is it being placed on, and is it configuring the memory for the RAZ instance as well?

Reply:
This hostGroup has "ranger-RANGER_ADMIN-BASE" - not RAZ.

There are NO references to RAZ in the template. How is it installed ?

>  in that case, the cm memory configuration will be enough
CM memory config will not run on a hostgroup which starts with 0 nodes.

Reply:
In medium-duty, RAZ is added by config providers in CB code and not by blueprint.
As RAZ is controlled by configuration it is not part of the blueprint.

we will need additional changes to it.

Reply:
Blueprint is static and changes controlled by configuration are done through config providers. 
That should be the case even for RAZ.

Reply:
Please make the changes in this PR itself (instead of a separate jira). We're trying to check this in with multiple different issues - while this is extremely close to being complete.

There's at least the following.
1. RAZ code changes to make sure it is on the razhg
2. Separating out roleConfigGroups for the new hostGroups - (this should be no more than a couple of hours)
3. kafka broker memory configuration not being set explicitly

Reply:
RAZ changes and new roleConfigGroups for the new hostGroups are done.

Memory configurations will be done separately.